### PR TITLE
Fix/mp 281 cbg 1194

### PR DESCRIPTION
**Pull request by sridevi-15**
#### Fixes #.

- [x] Ran `flake8`
- [x] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:

- Added verification to check for pattern on sync gateway log on sgcollect
- Added verification to verify list of all expected files in sgcollect zip

### CODE REVIEW

File: testsuites/syncgateway/functional/tests/test_log_redaction.py [hunk 1]
```diff
@@ -370,6 +370,73 @@ def test_sgCollectRestApi_errorMessages(params_from_base_test_setup, remove_tmp_
...
+    cluster = Cluster(config=cluster_config)
+    cluster.reset(sg_config_path=sg_conf)
+
+    # Get sync_gateway host
+    cluster = load_cluster_config_json(cluster_config)
+    sg_host = cluster["sync_gateways"][0]["ip"]
+    if cluster["environment"]["ipv6_enabled"]:
+        sg_host = "[{}]".format(sg_host)
+
+    # 3. Create user in sync_gateway
```
Comment (euniceatcouchbase) [inline testsuites/syncgateway/functional/tests/test_log_redaction.py:409]: you may remove #3, #4-ish, somehow they look too consistent with the 1 and 2 in the summary. :)

**Review by euniceatcouchbase — APPROVED** [review]
approved with a comment.

File: testsuites/syncgateway/functional/tests/test_log_redaction.py [hunk 2]
```diff
@@ -370,6 +370,73 @@ def test_sgCollectRestApi_errorMessages(params_from_base_test_setup, remove_tmp_
...
+    if get_sync_gateway_version(sg_ip)[0] < "2.1":
+        pytest.skip("log redaction feature not available for version < 2.1 ")
+
+    sg_conf = sync_gateway_config_path_for_mode(sg_conf_name, mode)
+
+    cluster = Cluster(config=cluster_config)
+    cluster.reset(sg_config_path=sg_conf)
+
+    # Get sync_gateway host
+    cluster = load_cluster_config_json(cluster_config)
```
Comment (mgcouchbase) [inline testsuites/syncgateway/functional/tests/test_log_redaction.py:404]: I think you can get the host from ,cluster_hosts variable ?

File: testsuites/syncgateway/functional/tests/test_log_redaction.py [hunk 3]
```diff
@@ -370,6 +370,73 @@ def test_sgCollectRestApi_errorMessages(params_from_base_test_setup, remove_tmp_
...
+
+    # 4. Create docs with xattrs
+    sgdoc_bodies = document.create_docs(doc_id_prefix='sg_docs', number=num_of_docs,
+                                        attachments_generator=attachment.generate_2_png_10_10, channels=channels)
+    sg_client.add_bulk_docs(url=sg_url, db=sg_db, docs=sgdoc_bodies, auth=autouser_session)
+    assert len(sgdoc_bodies) == num_of_docs
+
+    resp = sg_client.sgCollect_info(sg_host)
+    if resp["status"] != "started":
+        assert False, "sg collect did not started"
```
Comment (mgcouchbase) [inline testsuites/syncgateway/functional/tests/test_log_redaction.py:423]: assert resp["status"] != "started",  "sg collect did not started"

File: testsuites/syncgateway/functional/tests/test_log_redaction.py (hunk 2 above)
Comment (mgcouchbase) [reply testsuites/syncgateway/functional/tests/test_log_redaction.py:404]: cluster_conf = params_from_base_test_setup["cluster_config"],  get_sg_version(cluster_conf) > 2.5

**Review by mgcouchbase — APPROVED** [review]

File: testsuites/syncgateway/functional/tests/test_log_redaction.py (hunk 2 above)
Comment (sridevi-15) [reply testsuites/syncgateway/functional/tests/test_log_redaction.py:403]: Getting from cluster_hosts does not return ip, it returns full url. you need to write seperate method to chop the url and get ip, instead as well it is okay to use load_clust_config_json which does not take much time

File: testsuites/syncgateway/functional/tests/test_log_redaction.py (hunk 3 above)
Comment (sridevi-15) [reply testsuites/syncgateway/functional/tests/test_log_redaction.py]: done

File: testsuites/syncgateway/functional/tests/test_log_redaction.py (hunk 1 above)
Comment (sridevi-15) [reply testsuites/syncgateway/functional/tests/test_log_redaction.py]: done

**Pull request merged**